src/libraries/System.Net.Security/src/System/Net/Security/MD4.cs

```diff
@@ -0,0 +1,271 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+
+namespace System.Net.Security
+{
+    internal sealed class MD4
+    {
+        private byte[]? HashValue;
+        private uint[] state;
+        private byte[] buffer;
+        private uint[] count;
+        private uint[] x;
+
+        private const int S11 = 3;
+        private const int S12 = 7;
+        private const int S13 = 11;
+        private const int S14 = 19;
+        private const int S21 = 3;
+        private const int S22 = 5;
+        private const int S23 = 9;
+        private const int S24 = 13;
+        private const int S31 = 3;
+        private const int S32 = 9;
+        private const int S33 = 11;
+        private const int S34 = 15;
+
+        private byte[] digest;
+        public int HashSize => 128;
+        public int HashSizeBytes => HashSize / 8;
+
+        public MD4()
+        {
+            // we allocate the context memory
+            state = new uint[4];
+            count = new uint[2];
+            buffer = new byte[64];
+            digest = new byte[16];
+            // temporary buffer in MD4Transform that we don't want to keep allocate on each iteration
+            x = new uint[16];
+            // the initialize our context
+            Initialize();
+        }
+
+        public int HashData(ReadOnlySpan<byte> source, Span<byte> destination)
+        {
+            if (destination.Length < HashSizeBytes)
+            {
+                throw new ArgumentException("Destination is too short.", nameof(destination));
+            }
+
+            ComputeHash(source.ToArray(), 0, source.Length).CopyTo(destination);
+            return destination.Length;
+        }
+
+        private void Initialize()
+        {
+            count[0] = 0;
+            count[1] = 0;
+            state[0] = 0x67452301;
+            state[1] = 0xefcdab89;
+            state[2] = 0x98badcfe;
+            state[3] = 0x10325476;
+            // Zeroize sensitive information
+            Array.Clear(buffer, 0, 64);
+            Array.Clear(x, 0, 16);
+        }
+
+        private byte[] ComputeHash(byte[] array, int offset, int count)
+        {
+            HashCore(array, offset, count);
+            HashValue = HashFinal();
+            byte[] result = (byte[]) HashValue.Clone();
+            Initialize();
```

Comment:
Why is this needed? Isn't `Initialize()` called at the end of `HashFinal()` already?

Reply:
no longer used, thanks